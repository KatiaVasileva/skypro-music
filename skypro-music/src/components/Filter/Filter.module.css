.filter {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin-bottom: 51px;
    position: relative;
}

.filterTitle {
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    margin-right: 15px;
}

.filterButton, .filterButtonActive {
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    border: 1px solid #ffffff;
    border-radius: 60px;
    padding: 6px 20px;
    /* position: relative; */

    &:not(:last-child) {
        margin-right: 10px;
    }

    &:hover {
        border-color: #d9b6ff;
        color: #d9b6ff;
        cursor: pointer;
    }

    &:active {
        border-color: #ad61ff;
        color: #ad61ff;
        cursor: pointer;
    }
}

.filterButtonActive {
    color: #ad61ff;
    border-color: #ad61ff;
}

.selectedPerformerFilterCount, .selectedYearFilterCount, .selectedGenreFilterCount {
    position: absolute;
    top: -10px;
    left: 240px;
    width: 26px;
    height: 26px;
    text-align: center;
    background-color: #6b46c1;
    color: #fff;
    border-radius: 50%;
    padding: 2px 8px;
    font-size: 13px;
    font-weight: bold;
}

.selectedYearFilterCount {
    left: 400px;
}

.selectedGenreFilterCount {
    left: 505px;
}

.skeletonContainer {
    display: flex;
    flex-direction: row;
    gap: 5px;
}