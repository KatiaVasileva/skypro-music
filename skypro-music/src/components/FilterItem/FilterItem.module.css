.popup {
    position: relative;
}

.popupContainer {
    position: absolute;
    width: 248px;
    max-height: 305px;
    background-color: #313131;
    border-radius: 12px;
    display: flex;
    margin-top: -40px;
    z-index: 1;
}

.popupContainerAuthor {
    margin-left: 100px;
}

.popupContainerYear {
    margin-left: 262px;
    height: 196px;
    width: 221px;
}

.popupContainerGenre {
    margin-left: 425px;
}

.popupBox {
    width: 180px;
    max-height: 237px;
    display: flex;
    margin-top: 20px;
    overflow-y: hidden;
    overflow-y: auto;
    margin: 34px 34px;
    scrollbar-color: #ffffff #4B4949;
    scrollbar-width: thin;
    min-height: 128px;
    min-width: 153px;
}

.popupContent {
    width: 152px;
    max-height: 232px;
    display: flex;
    flex-direction: column;
    gap: 24px;
    min-height: 128px;
}


.popupText {
    font-size: 18px;
    font-weight: 400;
    line-height: 24px;

    &:hover {
        color: #B672FF;
        text-decoration: underline;
        cursor: pointer;
    }
}