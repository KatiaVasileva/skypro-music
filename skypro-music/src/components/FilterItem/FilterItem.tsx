import styles from "./FilterItem.module.css";
import classNames from "classnames";

type FilterItemProps = {
  filterName: string;
  filterContent: Array<string>;
  selectedValues: Array<string>;
  handleFilterItemClick: (filterElement: string) => void;
  handleFilterElementClick?: () => void;
  isFilterElementClicked?: boolean;

};

function FilterItem({
  filterName,
  filterContent,
  selectedValues,
  handleFilterItemClick,
  handleFilterElementClick,
  isFilterElementClicked,
}: FilterItemProps) {

  return (
    <div className={styles.popup}>
      <div
        className={classNames(
          styles.popupContainer,
          filterName === "performer"
            ? styles.popupContainerAuthor
            : filterName === "year"
            ? styles.popupContainerYear
            : styles.popupContainerGenre
        )}
      >
        <div className={styles.popupBox}>
          <div className={styles.popupContent}>
            {filterContent.map((filterElement, index) => (
              <p
                className={classNames(
                  styles.popupText,
                    {
                      [styles.popupTextActive]:
                        selectedValues.includes(filterElement),
                      [styles.popupTextInactive]:
                        filterName === "year" && 
                        !selectedValues.includes(filterElement) && isFilterElementClicked,
                    },
                )}
                key={index}
                onClick={() => {
                  handleFilterItemClick(filterElement);
                  if (handleFilterElementClick) {
                    handleFilterElementClick();
                  }
                  console.log(isFilterElementClicked);
                }}
              >
                {filterElement}
              </p>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
}

export default FilterItem;
