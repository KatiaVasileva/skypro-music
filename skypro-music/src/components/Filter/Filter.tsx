"use client";

import { useState } from "react";
import styles from "./Filter.module.css";
import { FilterProps } from "@/types/FilterProps.types";
import FilterItem from "../FilterItem/FilterItem";
import { useAppDispatch, useAppSelector } from "@/store/store";
import { setDateState, setGenreState, setPerformerState } from "@/store/features/filterSlice";

function Filter({ performers, genres, years }: FilterProps) {
  const [activeIndex, setActiveIndex] = useState<number>(0);
  const dispatch = useAppDispatch();
  const {performerState, dateState, genreState} = useAppSelector((state) => state.filter);

  let dateOrderArr: Array<string> = [];
  dateOrderArr.push(dateState);

  const handleFilterItemClick = (filterElement: string) => {
    if (activeIndex === 1) {
      dispatch(setPerformerState(filterElement));
    }
    if (activeIndex === 2) {
      dispatch(setDateState(filterElement));
    }
    if (activeIndex === 3) {
      dispatch(setGenreState(filterElement));
    }
  };

  return (
    <>
      <div className={styles.filter}>
        <div className={styles.filterTitle}>Искать по:</div>
        <div
          className={
            activeIndex === 1 ? styles.filterButtonActive : styles.filterButton
          }
          onClick={() =>
            activeIndex === 1 ? setActiveIndex(0) : setActiveIndex(1)
          }
        >
          исполнителю
        </div>
        <div
          className={
            activeIndex === 2 ? styles.filterButtonActive : styles.filterButton
          }
          onClick={() =>
            activeIndex === 2 ? setActiveIndex(0) : setActiveIndex(2)
          }
        >
          году выпуска
        </div>
        <div
          className={
            activeIndex === 3 ? styles.filterButtonActive : styles.filterButton
          }
          onClick={() =>
            activeIndex === 3 ? setActiveIndex(0) : setActiveIndex(3)
          }
        >
          жанру
        </div>
      </div>

      {activeIndex === 1 && (
        <FilterItem
          filterName="performer"
          filterContent={performers}
          selectedValues={performerState}
          selectedFilterCount={performerState.length}
          handleFilterItemClick={handleFilterItemClick}
        />
      )}

      {activeIndex === 2 && (
        <FilterItem
          filterName="year"
          filterContent={years}
          selectedValues={dateOrderArr}
          selectedFilterCount={dateOrderArr.length}
          handleFilterItemClick={handleFilterItemClick}
        />
      )}

      {activeIndex === 3 && (
        <FilterItem
          filterName="genre"
          filterContent={genres}
          selectedValues={genreState}
          selectedFilterCount={genreState.length}
          handleFilterItemClick={handleFilterItemClick}
        />
      )}
    </>
  );
}

export default Filter;
