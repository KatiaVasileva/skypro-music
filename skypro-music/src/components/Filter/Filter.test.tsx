import ReduxProvider from "@/store/ReduxProvider";
import { render } from "@testing-library/react";
import Filter from "./Filter";
import "@testing-library/jest-dom";
import FilterItem from "../FilterItem/FilterItem";

describe("Filter component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ReduxProvider>
        <Filter performers={[]} genres={[]} years={[]}          
        />
      </ReduxProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
