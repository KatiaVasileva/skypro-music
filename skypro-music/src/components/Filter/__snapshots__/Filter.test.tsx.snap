// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders correctly 1`] = `
<div>
  <div
    class="filter"
  >
    <div
      class="filterTitle"
    >
      Искать по:
    </div>
    <div
      class="filterButton"
    >
      исполнителю
    </div>
    <div
      class="filterButton"
    >
      году выпуска
    </div>
    <div
      class="filterButton"
    >
      жанру
    </div>
  </div>
</div>
`;
